src/components/Navigation.js

```diff
@@ -0,0 +1,101 @@
+import styled from "styled-components";
+import { useRouter } from "next/router";
+
+export default function Navigation() {
+  const router = useRouter();
+  const { index } = router.query;
+  const currentPath = router.pathname;
+
+  function handleDownstairs() {
+    if (currentPath === "/boss-floor") {
+      router.push("/floor/2");
+    } else if (index === "1") {
+      return;
+    } else {
+      router.push(`/floor/${parseInt(index, 10) - 1}`);
+    }
+  }
+
+  function handleUpstairs() {
+    if (currentPath === "/boss-floor") {
+      return;
+    } else if (index === "2") {
+      router.push("/boss-floor");
+    } else {
+      router.push(`/floor/${parseInt(index, 10) + 1}`);
+    }
+  }
+
+  function ordinalSuffix(i) {
+    const j = i % 10;
+    const k = i % 100;
+    if (j === 1 && k !== 11) {
+      return i + "st";
+    }
+    if (j === 2 && k !== 12) {
+      return i + "nd";
+    }
+    if (j === 3 && k !== 13) {
+      return i + "rd";
+    }
+    return i + "th";
+  }
```

Comment:
Weird and cute at the same time. 🐰
I would try to write it dynamically in case another floor is added in the future. But for this moment it works perfectly. 

Nevertheless, I would choose the variable names differently, especially `i` - which probably stands for `important`. 
Just kidding - but it is not immediately clear that the index is meant.

Reply:
I have this part from copilot, and it is actually already dynamic. -> ex. 72nd 73rd 74th.

i stands for integer here, it has the same meaning like the i in foor loops.